zap/src/main/java/org/zaproxy/zap/extension/ascan/CustomScanDialog.java

```diff
@@ -864,6 +882,20 @@ public void save() {
 
             contextSpecificObjects.add(scannerParam);
             contextSpecificObjects.add(techTreeState);
+            try {
+                List<ScanFilter> scanFilterList = filterPanel.getScanFilters();
+                for (ScanFilter scanFilter : scanFilterList) {
+                    contextSpecificObjects.add(scanFilter);
+                }
+            } catch (PatternSyntaxException e1) {
```

Comment:
We should validate these beforehand, in `validateFields` (in this method everything should just work).

Reply:
actual issue is on clicking `scan` in Parent panel (Active Scan) we are getting filters and also in case url pattern is having issues we need to stop the scan so i tried multiple ways but found no way except throwing it to parent and then stoping parent from running scan.

Reply:
Yeah, I understand the need but we should do it earlier, the filter panel can have a validate method for that purpose, which is called from `validateFields`.

Reply:
oh .. didn't know that there was some method for this. thanks.